src/test.rs

```diff
@@ -98,7 +132,163 @@ async fn wallet_is_persisted() -> anyhow::Result<()> {
+    let mut wallet = Wallet::create(external_desc.to_string(), internal_desc.to_string())
+        .network(NETWORK)
+        .create_wallet_async(&mut store)
+        .await?;
+    let _ = electrum_full_scan(&mut wallet).await?;
+    assert!(wallet.persist_async(&mut store).await?);
+    Ok((store, wallet_name))
+}
+
+async fn load_wallet_and_get_transactions(
+    store: &mut Store,
+    external_desc: &str,
+    internal_desc: &str,
+) -> anyhow::Result<Vec<Txid>> {
+    let wallet = Wallet::load()
+        .descriptor(KeychainKind::External, Some(external_desc.to_string()))
+        .descriptor(KeychainKind::Internal, Some(internal_desc.to_string()))
+        .load_wallet_async(store)
+        .await?
+        .expect("wallet must exist");
+    Ok(wallet.transactions().map(|tx| tx.tx_node.txid).collect())
+}
+
+async fn recover_wallet_and_get_transactions(
+    external_desc: &str,
+    internal_desc: &str,
+) -> anyhow::Result<Vec<Txid>> {
+    let mut wallet = Wallet::create(external_desc.to_string(), internal_desc.to_string())
+        .network(NETWORK)
+        .create_wallet_no_persist()?;
+    let _ = electrum_full_scan_no_persist(&mut wallet).await?;
+    Ok(wallet.transactions().map(|tx| tx.tx_node.txid).collect())
+}
+
+#[tracing::instrument]
+#[tokio::test]
+async fn test_three_wallets_list_transactions() -> anyhow::Result<()> {
+    Settings::debug()
+        .most_recent_first(false)
+        .lineno_suffix(true)
+        .install();
+    default_provider()
+        .install_default()
+        .expect("Failed to install rustls default crypto provider");
+
+    let url = setup_database().await?;
+
+    let wallet_types = [1, 2, 3];
+    let mut stores = Vec::new();
+    let mut persisted_txs = Vec::new();
+    let mut recovered_txs = Vec::new();
+
+    for wallet_type in wallet_types.iter() {
+        let (external_desc, internal_desc) = get_wallet_descriptors(*wallet_type);
+        let (store, _) = create_and_scan_wallet(&url, external_desc, internal_desc).await?;
+        stores.push(store);
+    }
+
+    for (i, store) in stores.iter_mut().enumerate() {
+        let (external_desc, internal_desc) = get_wallet_descriptors(wallet_types[i]);
+        let mut txs = load_wallet_and_get_transactions(store, external_desc, internal_desc).await?;
+        txs.sort();
+        persisted_txs.push(txs);
+    }
+
+    for wallet_type in wallet_types.iter() {
+        let (external_desc, internal_desc) = get_wallet_descriptors(*wallet_type);
+        let mut txs = recover_wallet_and_get_transactions(external_desc, internal_desc).await?;
+        txs.sort();
+        recovered_txs.push(txs);
+    }
+
+    for i in 0..3 {
+        assert_eq!(persisted_txs[i], recovered_txs[i]);
+    }
+
+    // Clean up
+    let db = PgPool::connect(&url).await?;
+    drop_all(db).await.expect("hope it's not mainnet");
+    Ok(())
+}
+
+async fn electrum_full_scan(wallet: &mut PersistedWallet<Store>) -> anyhow::Result<()> {
```

Comment:
I'd rather not having unit tests that depend on external network services like mempool.space since the tests will be flaky if mempool.space rate limits tests.  Also even though DB tests are more of integration tests, they should be focused as much as possible just on the code being tested plus the local postgres DB. 

But if you don't have time to create more self contained tests right now I'm ok with merging this as-is as long as we rework the tests in a future PR.